File: striver/S1L4/java/Divisors.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Collections;

public class Divisors {
    public static ArrayList<Integer> printDivisors(int num) {
        ArrayList<Integer> divisors = new ArrayList<>();
        for (int i = 1; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                divisors.add(i);
                if (i != num / i) {  // Avoid duplicate for perfect squares
                    divisors.add(num / i);
                }
            }
        }
        Collections.sort(divisors);
        return divisors;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int num = sc.nextInt();
        ArrayList<Integer> divisors = printDivisors(num);
        System.out.println("Divisors of " + num + " are: " + divisors);
        sc.close();
    }
} 

File: striver/S1L4/python/divisors.py
def print_divisors(num):
    """
    Print all divisors of a number
    Example: 12 -> 1, 2, 3, 4, 6, 12
    """
    divisors = []
    for i in range(1, int(num ** 0.5) + 1):
        if num % i == 0:
            divisors.append(i)
            if i != num // i:  # Avoid duplicate for perfect squares
                divisors.append(num // i)
    return sorted(divisors)

if __name__ == "__main__":
    num = int(input("Enter a number: "))
    divisors = print_divisors(num)
    print(f"Divisors of {num} are: {divisors}") 